### PR TITLE
[NCL-6260]: Create a method to read Orch message and respond once analysis is finished

### PR DESCRIPTION
**Pull request by jbartece**

### CODE REVIEW

File: src/main/java/org/jboss/pnc/deliverablesanalyzer/Finder.java [hunk 1]
```diff
@@ -174,4 +248,20 @@ private FinderResult findBuilds(
...
+
+    /*
+     * { // Issues - throw exception from lambda, support cancel List<CompletableFuture<FinderResult>> futureResults =
+     * new ArrayList(); for (String rawUrl : urls) { futureResults.add(executor.supplyAsync(() -> { URL url =
+     * URI.create(rawUrl).normalize().toURL(); return find(id, url, distributionAnalyzerListener, buildFinderListener,
+     * config); })); }
+     *
+     * for (CompletableFuture<FinderResult> future : futureResults) { try { results.add(future.get()); } catch
+     * (InterruptedException | ExecutionException e) { e.printStackTrace(); } } }
+     */
```
Comment (janinko) [inline src/main/java/org/jboss/pnc/deliverablesanalyzer/Finder.java:266]: https://www.youtube.com/embed/EVeEIBBADxw?start=11&end=19&version=3

File: src/main/java/org/jboss/pnc/deliverablesanalyzer/Finder.java [hunk 2]
```diff
@@ -123,19 +171,21 @@ public FinderResult find(
...
+            DistributionAnalyzer analyzer = new DistributionAnalyzer(files, config, cacheManager);
             analyzer.setListener(distributionAnalyzerListener);
 
             Future<Map<ChecksumType, MultiValuedMap<String, String>>> futureChecksum = pool.submit(analyzer);
             result = findBuilds(id, url, analyzer, futureChecksum, buildFinderListener);
 
             LOGGER.info("Done finding builds for {}", url);
         } finally {
-            boolean isClean = cleanup(config.getOutputDirectory());
+            int a = 1 + 2;
```
Comment (dwalluck) [inline src/main/java/org/jboss/pnc/deliverablesanalyzer/Finder.java:185]: OK?

Comment (jbartece) [reply src/main/java/org/jboss/pnc/deliverablesanalyzer/Finder.java:185]: Maven plugins didn't let me have the empty block. I'll fix it. It is just temporary :-)

**Comment by jbartece** [comment]
I need to do a few things: Implement HttpClient, write tests, try to make more readable the parallelism in Finder and fix cleanup of the resource.

**Comment by jbartece** [comment]
The move of the DTOs to pnc-api introduces dependency on build-finder and it is not easy to remove it - it would require to move the build-finder classes too. After discussion, we have decided to close that PR and keep the classes in DelA

**Comment by jbartece** [comment]
I'd like to get this work merged and the following ones postpone:

- Review/fix cleanup of resources
- Take a look at the HeartBeatConfig from pnc-api
- Check MDC headers and the work related, Add it to the cancel request

I'll probably create issues for those, because I won't be able to work on them this/next week.

**Comment by jbartece** [comment]
retest this please

**Comment by jbartece** [comment]
I'm merging the work to unblock other progress. I will raise issues for the following work:

- Review/fix cleanup of resources
- Take a look at the HeartBeatConfig from pnc-api
- Check MDC headers and the work related, Add it to the cancel request

**Pull request merged**